src/summarizedexperiment/BaseSE.py

```diff
@@ -506,3 +509,209 @@ def toAnnData(
+
+    def _create_samples_if_missing(self, sample_names: Sequence[str], df: DataFrame):
+        """Create a new sample populated with nans if it doesn't exist.
+
+        Args:
+            sample_names (str): List of sample names that should exist.
+            df (DataFrame): The dataframe.
+        """
+        for sample_name in sample_names:
+            if sample_name not in df.columns:
+                df[sample_name] = np.nan
+
+    def _create_features_if_missing(
+        self, feature_names: Sequence[str], df: DataFrame
+    ) -> DataFrame:
+        """Create a new feature populated with nans if it doesn't exist.
+
+        Args:
+            feature_names (str): List of feature names that should exist.
+            df (DataFrame): The dataframe.
+
+        Returns:
+            DataFrame: Assay with missing features added.
+        """
+        all_features = df.index.union(feature_names, sort=False)
+        return df.reindex(index=all_features)
+
+    def combineCols(
+        self, *summarized_experiments: "BaseSE", use_names: bool = True, fill=np.nan
+    ) -> "BaseSE":
+        """A more flexible version of `cbind`. Permits differences in the number and identity of rows,
+        differences in `colData` fields, and even differences in the available `assays` among
+        `SummarizedExperiment` objects being combined.
+
+        Only considering RNA-Seq experiments for now.
+
+        Args:
+            summarized_experiments ("BaseSE"): `SummarizedExperiment` objects to concatenate.
+            use_names (bool):
+                If `True`, then each input `SummarizedExperiment` must have non-null, non-duplicated row names.
+                The row names of the resultant `SummarizedExperiment` object will be the union of the row
+                names across all input objects.
+                If `False`, then each input `SummarizedExperiment` object must have the same number of rows.
+                The row names of the resultant `SummarizedExperiment` object will simply be the row names of
+                the first `SummarizedExperiment`.
+            fill (scalar value):
+                The value to fill NaNs.
+
+        Raises:
+            TypeError: if any of the provided objects are not "SummarizedExperiment".
+            ValueError: if not all assays have the same dimensions.
+
+        Returns:
+            BaseSE: new concatenated `SummarizedExperiment` object.
+        """
+
+        self._validate_objects(summarized_experiments)
+
+        ses = [self] + list(summarized_experiments)
+
+        self._validate_shapes(ses)
+
+        new_metadata = {}
+        for se in ses:
+            if se.metadata:
+                new_metadata.update(se.metadata.copy())
+
+        rowDatas = []
+        colDatas = []
+        for se in ses:
+            rowDatas.append(se.rowData.copy())
+            colDatas.append(se.colData.copy())
+
+        new_colData = reduce(lambda left, right: pd.concat([left, right]), colDatas)
```

Comment:
assuming that `rowDatas` can be a mix of `BiocFrame` and `pd.DataFrame` objects, could we simply convert `BiocFrame` objects to pandas before appending in lines 655/656?

Reply:
> `use_names` or not, the underlying abstract code to merge `DataFrames` or `BiocFrames` should be the same between `coldata` and `rowdata`.

not exactly sure what you mean by this - are you just talking about dispatching on the type of dataframe?

Reply:
> assuming that rowDatas can be a mix of BiocFrame and pd.DataFrame objects, could we simply convert BiocFrame objects to pandas before appending in lines 655/656?

I am thinking of scenarios where you don't have to do that. 

Pandas is restricted to the dtypes for each column, hence one reason why BiocFrame exists because it supports awkward column types => nested, irregular or any **_array-like_** representation that implements the `Sequence` interface. 

I would rather convert the pandas dataframe into a columnar representation. 

When we want to merge two dataframes, there's really only two options

- by index (this requires all dataframes to contain an index) or
- by row (this requires both frames to have the same number of rows)

If we implement this operation, we can reuse the same logic for both row data and column data.

Reply:
`use.names` is specifying where we merge by index or by row.